### ISSUE TITLE
All authentication plugins should require admin login to be used on the form-side

### ISSUE THREAD

**Issue by joeribekker** [open-formulieren/open-forms#1322]
This prevents any unwanted or accidental use of the test plugins by unauthorized persons.

**Tasks**

* [x] Require all "test" plugins to have an active admin session before they can be used on a form
* [x] Mark "DigiD simulatie" as test plugin

**Comment by sergei-maertens** [comment]
#805 explicitly made us mark some demo/test plugins as "not-demo", which makes it they can be used by non-staff users. This ticket requires us to mark them as demo again, which contradicts the original ticket.

In my opinion, undoing #805 is fine, since demo plugins can be enabled via the feature flags for testing. And with #1384, it's possible to enable demo plugins through the feature flag, and disable a subset of them.

tl;dr - in my opinion we can mark the kvk/bsn plugins as demo again
